gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/common/MongoFactory.java

```diff
@@ -136,31 +137,30 @@ private ServerSettings buildServerSettings() {
         return serverBuilder.build();
     }
 
-    private SslSettings buildSslSettings() {
+    private SslSettings buildSslSettings() throws NoSuchAlgorithmException {
         SslSettings.Builder sslBuilder = SslSettings.builder();
 
-        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class);
-
-        if (sslEnabled != null) {
-            sslBuilder.enabled(sslEnabled);
-            if (sslEnabled.booleanValue()) {
-                String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
-                String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
-                String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
-
-                if (keystore != null) {
-                    try {
-                        SSLContext ctx = SSLContext.getInstance("TLS");
-                        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-                        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-                        ks.load(new FileInputStream(keystore), keystorePassword.toCharArray());
-                        keyManagerFactory.init(ks, keyPassword.toCharArray());
-                        ctx.init(keyManagerFactory.getKeyManagers(), null, null);
-                        sslBuilder.context(ctx);
-                    } catch (Exception e) {
-                        logger.error(e.getCause().toString());
-                        throw new IllegalStateException("Error creating the keystore for mongodb", e);
-                    }
+        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class, false);
+        sslBuilder.enabled(sslEnabled);
+
+        if (sslEnabled) {
+            String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
+            String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
+            String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
+            String tlsProtocol = readPropertyValue(propertyPrefix + "tlsProtocol", String.class);
+
+            if (keystore != null) {
+                try {
+                    SSLContext ctx = tlsProtocol == null ? SSLContext.getDefault() : SSLContext.getInstance(tlsProtocol);
```

Comment:
A user could downgrade the default TLS version used by the JVM.
Even without knowing it, for example, a user upgrade his JVM without thinking to change the setting that was previously set.
Is that a vulnerability ?

Anyway, it sounds tricky to make a comparison between SSLContext.getDefault().getProtocol() and the protocol from settings.

Reply:
For me that's ok as it's the user who's looking at downgrading TLS specifically.
So he knows what he does!